packages/dev/core/src/FlowGraph/Blocks/Event/flowGraphSceneReadyEventBlock.ts

```diff
@@ -1,38 +1,37 @@
-import type { Nullable } from "../../../types";
-import type { Observer } from "../../../Misc/observable";
 import type { Scene } from "../../../scene";
-import type { FlowGraph } from "../../flowGraph";
 import { FlowGraphEventBlock } from "../../flowGraphEventBlock";
+import type { FlowGraphContext } from "core/FlowGraph/flowGraphContext";
 
 /**
  * @experimental
  * Block that triggers when a scene is ready.
  */
 export class FlowGraphSceneReadyEventBlock extends FlowGraphEventBlock {
-    private _scene: Scene;
-    private _sceneReadyObserver: Nullable<Observer<Scene>>;
-
-    public constructor(graph: FlowGraph, scene: Scene) {
-        super(graph);
-        this._scene = scene;
+    public constructor() {
+        super();
     }
 
     /**
      * @internal
      */
-    public _startListening(): void {
-        if (!this._sceneReadyObserver) {
-            this._sceneReadyObserver = this._scene.onReadyObservable.add(() => {
-                this._execute();
+    public _preparePendingTasks(context: FlowGraphContext): void {
+        let contextObserver = context._getExecutionVariable(this, "sceneReadyObserver");
+        if (!contextObserver) {
+            const scene = context._getGraphVariable("scene") as Scene;
+            contextObserver = scene.onReadyObservable.add(() => {
+                this._execute(context);
             });
+            context._setExecutionVariable(this, "sceneReadyObserver", contextObserver);
         }
     }
 
     /**
      * @internal
      */
-    public _stopListening() {
-        this._scene.onReadyObservable.remove(this._sceneReadyObserver);
-        this._sceneReadyObserver = null;
+    public _cancelPendingTasks(context: FlowGraphContext) {
+        const contextObserver = context._getExecutionVariable(this, "sceneReadyObserver");
+        const scene = context._getGraphVariable("scene") as Scene;

```

Comment:
Yep, the scene graph variable should always be set. I could change the context to have a dictionary of graph variables, so that they're always set? Something like

```
graphVariables = {scene: Scene}
```

and we access graphVariables directly.